nav2_collision_monitor/src/polygon.cpp

```diff
@@ -0,0 +1,376 @@
+      return false;
+    }
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".max_points", rclcpp::ParameterValue(3));
+    max_points_ = node->get_parameter(polygon_name_ + ".max_points").as_int();
+
+    if (action_type_ == SLOWDOWN) {
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + ".slowdown_ratio", rclcpp::ParameterValue(0.5));
+      slowdown_ratio_ = node->get_parameter(polygon_name_ + ".slowdown_ratio").as_double();
+    }
+
+    if (action_type_ == APPROACH) {
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + ".time_before_collision", rclcpp::ParameterValue(2.0));
+      time_before_collision_ =
+        node->get_parameter(polygon_name_ + ".time_before_collision").as_double();
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + ".simulation_time_step", rclcpp::ParameterValue(0.02));
+      simulation_time_step_ =
+        node->get_parameter(polygon_name_ + ".simulation_time_step").as_double();
+    }
+
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".visualize", rclcpp::ParameterValue(false));
+    visualize_ = node->get_parameter(polygon_name_ + ".visualize").as_bool();
+    if (visualize_) {
+      // Get polygon topic parameter in case if it is going to be published
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + ".polygon_topic", rclcpp::ParameterValue(polygon_name_));
+      polygon_topic = node->get_parameter(polygon_name_ + ".polygon_topic").as_string();
+    }
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(
+      logger_,
+      "[%s]: Error while getting common polygon parameters: %s",
+      polygon_name_.c_str(), ex.what());
+    return false;
+  }
+
+  return true;
+}
+
+bool Polygon::getParameters(std::string & polygon_topic, std::string & footprint_topic)
+{
+  auto node = node_.lock();
+  if (!node) {
+    throw std::runtime_error{"Failed to lock node"};
+  }
+
+  if (!getCommonParameters(polygon_topic)) {
+    return false;
+  }
+
+  try {
+    if (action_type_ == APPROACH) {
+      // Obtain the footprint topic to make a footprint subscription for approach polygon
+      nav2_util::declare_parameter_if_not_declared(
+        node, polygon_name_ + ".footprint_topic", rclcpp::ParameterValue("footprint"));
+      footprint_topic =
+        node->get_parameter(polygon_name_ + ".footprint_topic").as_string();
+
+      // This is robot footprint: do not need to get polygon points from ROS parameters.
+      // It will be set dynamically later.
+      return true;
+    } else {
+      // Make it empty otherwise
+      footprint_topic.clear();
+    }
+
+    // Leave it not initialized: the will cause an error if it will not set
+    nav2_util::declare_parameter_if_not_declared(
+      node, polygon_name_ + ".points", rclcpp::PARAMETER_DOUBLE_ARRAY);
```

Comment:
Isn't this only relevent for non-approach style polygons? Right now this would throw an exception for the approach case because the `points` were not specified

Reply:
This chain of code will work only for non-approach cases. For the approach case we will never be here since previous part of code will make a return from function:
```
if (action_type_ == APPROACH) {
      // Obtain the footprint topic to make a footprint subscription for approach polygon
      nav2_util::declare_parameter_if_not_declared(
        node, polygon_name_ + ".footprint_topic", rclcpp::ParameterValue("footprint"));
      footprint_topic =
        node->get_parameter(polygon_name_ + ".footprint_topic").as_string();

      // This is robot footprint: do not need to get polygon points from ROS parameters.
      // It will be set dynamically later.
      return true;  // <- Here we are for approach
    } else {
      // Make it empty otherwise
      footprint_topic.clear();
    }

    // Leave it not initialized: the will cause an error if it will not set
    nav2_util::declare_parameter_if_not_declared(
      node, polygon_name_ + ".points", rclcpp::PARAMETER_DOUBLE_ARRAY);
   ...
```

Reply:
Got it- sorry I missed that